cat-launcher/src-tauri/src/mods/get_mod_activity.rs

```diff
@@ -0,0 +1,44 @@
+use std::path::Path;
+use reqwest::Client;
+
+use crate::infra::github::api::{get_latest_commit, GetLatestCommitError};
+use crate::mods::get_third_party_mod::{get_third_party_mod, GetThirdPartyModError};
+use crate::variants::GameVariant;
+
+#[derive(thiserror::Error, Debug)]
+pub enum GetModActivityError {
+    #[error("failed to get third party mod: {0}")]
+    GetThirdPartyMod(#[from] GetThirdPartyModError),
+
+    #[error("mod with id `{0}` has no github activity configured")]
+    NoGithubActivity(String),
+
+    #[error("failed to get latest commit: {0}")]
+    GetLatestCommit(#[from] GetLatestCommitError),
+
+    #[error("failed to parse github url `{0}`")]
+    ParseGithubUrl(String),
+}
+
+pub async fn get_mod_activity(
+    mod_id: &str,
+    game_variant: &GameVariant,
+    resource_dir: &Path,
+    http_client: &Client,
+) -> Result<Option<String>, GetModActivityError> {
+    let third_party_mod = get_third_party_mod(mod_id, game_variant, resource_dir).await?;
+
+    if third_party_mod.activity.activity_type == "github_commit" {
+        if let Some(github_url) = third_party_mod.activity.github {
+            let url_parts: Vec<&str> = github_url.split('/').collect();
```

Comment:
The current method of parsing the GitHub URL by splitting the string and accessing parts by index is brittle. It assumes a very specific URL format ([`https://github.com/owner/repo`](https://github.com/owner/repo)) and will fail with variations like `github.com/owner/repo` or if the URL has extra path segments.

A more robust approach would be to find `github.com/` and then parse the path components that follow it. This would handle different protocols and URL structures gracefully.